### PR TITLE
Create an application/service in the new environment and push configu…

### PR DESCRIPTION
**Pull request by varshab1210**
…ration to GitOps repository scenarios

**What type of PR is this?**

/kind enhancement

**What does this PR do / why we need it**:

https://github.com/redhat-developer/kam/issues/136

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

https://github.com/redhat-developer/kam/issues/136

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: test/e2e/features/basic.feature [hunk 1]
```diff
@@ -49,3 +49,22 @@ Feature: Basic test
         And application "dev-env" should be in "Synced" state
         And application "stage-env" should be in "Synced" state
         And application "cicd-app" should be in "Synced" state
+
+    Scenario: Create an Application/Service in the new Environment
+        Given gitops repository is created
+        When executing "kam service add  --env-name new-env --app-name app-bus --service-name bus  --git-repo-url $GIT_REPO_URL --pipelines-folder $GITOPS_PATH" succeeds
+        Then stderr should be empty
+        
+    Scenario: Commit and Push configuration to GitOps repoository
```
Comment (amitkrout) [inline test/e2e/features/basic.feature:58]: This scenario is not required. Please remove it

**Review by amitkrout — CHANGES_REQUESTED** [review]

**Comment by varshab1210** [comment]
I tested the pr locally and it went fine. I am not sure why its failing in CI. Can you please take a look @amitkrout 

```
$ make e2e
Log successfully started, logging into: /home/varshab/go/src/github.com/redhat-developer/kam/test/e2e/out/test-results/e2e_2021-8-10_13-13-39.log
Running e2e test in: /home/varshab/go/src/github.com/redhat-developer/kam/test/e2e/out/test-run
Working directory set to: /home/varshab/go/src/github.com/redhat-developer/kam/test/e2e/out/test-run
Before suite
The bash instance has been started and will be used for testing.
Before feature
Feature: Basic test
  Checks whether KAM top-level commands behave correctly.
Before scenario

  Scenario: KAM version                                    # features/basic.feature:5
    When executing "kam version" succeeds                  # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stderr should be empty                            # shell.go:302 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldBeEmpty
    And stdout should match "kam\sversion\sv\d+\.\d+\.\d+" # shell.go:326 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldMatch
After scenario
2021/08/10 13:13:39 unable to delete repository: Not Found
Before scenario

  Scenario: Execute KAM bootstrap command without --push-to-git=true flag                                                                                                                                                                                 # features/basic.feature:10
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL --image-repo $IMAGE_REPO --dockercfgjson $DOCKERCONFIGJSON_PATH --git-host-access-token $GIT_ACCESS_TOKEN --output bootstrapresources" succeeds # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stderr should be empty                                                                                                                                                                                                                           # shell.go:302 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldBeEmpty
After scenario
2021/08/10 13:13:43 unable to delete repository: Not Found
Before scenario

  Scenario: Execute KAM bootstrap command that overwite the custom output manifest path                                                                                                                                                                               # features/basic.feature:14
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL --image-repo $IMAGE_REPO --dockercfgjson $DOCKERCONFIGJSON_PATH --git-host-access-token $GIT_ACCESS_TOKEN --output bootstrapresources --overwrite" succeeds # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stderr should be empty                                                                                                                                                                                                                                       # shell.go:302 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldBeEmpty
After scenario
2021/08/10 13:13:53 unable to delete repository: Not Found
Before scenario

  Scenario: KAM bootstrap command should fail if any one mandatory flag --git-host-access-token is missing       # features/basic.feature:18
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL" fails # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then exitcode should not equal "0"                                                                           # shell.go:318 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldNotEqual
After scenario
2021/08/10 13:13:55 unable to delete repository: Not Found
Before scenario

  Scenario: Bringing the bootstrapped environment up                                                                                                                                                                                                                  # features/basic.feature:22
    Given gitops repository is created                                                                                                                                                                                                                                # kamsuite.go:163 -> /kamsuite.createRepository
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL --image-repo $IMAGE_REPO --dockercfgjson $DOCKERCONFIGJSON_PATH --git-host-access-token $GIT_ACCESS_TOKEN --output bootstrapresources --overwrite" succeeds # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "cd bootstrapresources" succeeds                                                                                                                                                                                                                   # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git init ." succeeds                                                                                                                                                                                                                              # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git add ." succeeds                                                                                                                                                                                                                               # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git commit -m 'Initial commit.'" succeeds                                                                                                                                                                                                         # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git remote add origin $GITOPS_REPO_URL" succeeds                                                                                                                                                                                                  # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git branch -M main" succeeds                                                                                                                                                                                                                      # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git push -u origin main" succeeds                                                                                                                                                                                                                 # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "cd .." succeeds                                                                                                                                                                                                                                   # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
After scenario
2021/08/10 13:14:09 Successfully deleted repository: "https://github.com/varshab1210/gitops.git"
Before scenario

  Scenario: Bringing the deployment infrastructure up                                                                                                                                                                                                        # features/basic.feature:34
    Given gitops repository is created                                                                                                                                                                                                                       # kamsuite.go:163 -> /kamsuite.createRepository
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL --image-repo $IMAGE_REPO --dockercfgjson $DOCKERCONFIGJSON_PATH --git-host-access-token $GIT_ACCESS_TOKEN --output bootstrap --overwrite" succeeds # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "cd bootstrap" succeeds                                                                                                                                                                                                                   # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git init ." succeeds                                                                                                                                                                                                                     # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git add ." succeeds                                                                                                                                                                                                                      # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git commit -m 'Initial commit.'" succeeds                                                                                                                                                                                                # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git remote add origin $GITOPS_REPO_URL" succeeds                                                                                                                                                                                         # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git branch -M main" succeeds                                                                                                                                                                                                             # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git push -u origin main" succeeds                                                                                                                                                                                                        # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "oc apply -k config/argocd/" succeeds                                                                                                                                                                                                     # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then login argocd API server                                                                                                                                                                                                                             # kamsuite.go:203 -> /kamsuite.loginArgoAPIServerLogin
    And Wait for "60" seconds                                                                                                                                                                                                                                # kamsuite.go:190 -> /kamsuite.waitForTime
    Then application "argo-app" should be in "Synced" state                                                                                                                                                                                                  # kamsuite.go:195 -> /kamsuite.applicationState
    And application "dev-app-taxi" should be in "Synced" state                                                                                                                                                                                               # kamsuite.go:195 -> /kamsuite.applicationState
    And application "dev-env" should be in "Synced" state                                                                                                                                                                                                    # kamsuite.go:195 -> /kamsuite.applicationState
    And application "stage-env" should be in "Synced" state                                                                                                                                                                                                  # kamsuite.go:195 -> /kamsuite.applicationState
    And application "cicd-app" should be in "Synced" state                                                                                                                                                                                                   # kamsuite.go:195 -> /kamsuite.applicationState
    Then executing "oc delete -k config/argocd"                                                                                                                                                                                                              # shell.go:203 -> github.com/code-ready/clicumber/testsuite.ExecuteCommand
    Then executing "cd .." succeeds                                                                                                                                                                                                                          # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
After scenario
2021/08/10 13:16:24 Successfully deleted repository: "https://github.com/varshab1210/gitops.git"
Before scenario

  Scenario: Create an Application/Service in the new Environment                                                                                                                                                                                                   # features/basic.feature:55
    Given gitops repository is created                                                                                                                                                                                                                             # kamsuite.go:163 -> /kamsuite.createRepository
    When executing "kam bootstrap --service-repo-url $SERVICE_REPO_URL --gitops-repo-url $GITOPS_REPO_URL --image-repo $IMAGE_REPO --dockercfgjson $DOCKERCONFIGJSON_PATH --git-host-access-token $GIT_ACCESS_TOKEN --output servicescenario --overwrite" succeeds # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "cd servicescenario" succeeds                                                                                                                                                                                                                   # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git init ." succeeds                                                                                                                                                                                                                           # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git add ." succeeds                                                                                                                                                                                                                            # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git commit -m 'Initial commit.'" succeeds                                                                                                                                                                                                      # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git remote add origin $GITOPS_REPO_URL" succeeds                                                                                                                                                                                               # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git branch -M main" succeeds                                                                                                                                                                                                                   # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "git push -u origin main" succeeds                                                                                                                                                                                                              # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then executing "oc apply -k config/argocd/" succeeds                                                                                                                                                                                                           # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then login argocd API server                                                                                                                                                                                                                                   # kamsuite.go:203 -> /kamsuite.loginArgoAPIServerLogin
    And Wait for "60" seconds                                                                                                                                                                                                                                      # kamsuite.go:190 -> /kamsuite.waitForTime
    Then application "argo-app" should be in "Synced" state                                                                                                                                                                                                        # kamsuite.go:195 -> /kamsuite.applicationState
    And application "dev-app-taxi" should be in "Synced" state                                                                                                                                                                                                     # kamsuite.go:195 -> /kamsuite.applicationState
    And application "dev-env" should be in "Synced" state                                                                                                                                                                                                          # kamsuite.go:195 -> /kamsuite.applicationState
    And application "stage-env" should be in "Synced" state                                                                                                                                                                                                        # kamsuite.go:195 -> /kamsuite.applicationState
    And application "cicd-app" should be in "Synced" state                                                                                                                                                                                                         # kamsuite.go:195 -> /kamsuite.applicationState
    Then executing "cd .."                                                                                                                                                                                                                                         # shell.go:203 -> github.com/code-ready/clicumber/testsuite.ExecuteCommand
    When executing "kam environment add --env-name new-env --pipelines-folder servicescenario" succeeds                                                                                                                                                            # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stderr should be empty                                                                                                                                                                                                                                    # shell.go:302 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldBeEmpty
    When executing "kam service add --env-name new-env --app-name app-bus --service-name bus --git-repo-url $GIT_REPO_URL --pipelines-folder servicescenario" succeeds                                                                                             # shell.go:230 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stderr should be empty                                                                                                                                                                                                                                    # shell.go:302 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldBeEmpty
After scenario
2021/08/10 13:18:03 Successfully deleted repository: "https://github.com/varshab1210/gitops.git"
After feature

Feature: Local test
  This feature file captures local test automation.
  Due to certain technical challenges in OpenShiftCI test infra
  we are keeping test scenario in local feature file for verifying the bits locally.
  Once the CI challenges are fixed, we move these test under basic tag.

Feature: Manual test
  This feature file captures only manual test steps.
  Due to certain technical challenges in OpenShiftCI test infra
  we are keeping few test scenario in manual feature file.
  Once the challenges are fixed, we automate these manual steps too.
After suite

7 scenarios (7 passed)
60 steps (60 passed)
4m23.723375642s
ok      github.com/redhat-developer/kam/test/e2e        263.731s

```

File: test/e2e/features/basic.feature [hunk 2]
```diff
@@ -49,3 +49,29 @@ Feature: Basic test
...
+        And Wait for "120" seconds
+        Then application "argo-app" should be in "Synced" state
+        And application "dev-app-taxi" should be in "Synced" state
+        And application "dev-env" should be in "Synced" state
+        And application "stage-env" should be in "Synced" state
+        And application "cicd-app" should be in "Synced" state
+        Then executing "cd .."
+        When executing "kam environment add --env-name new-env --pipelines-folder servicescenario" succeeds
+        Then stderr should be empty
+        When executing "kam service add --env-name new-env --app-name app-bus --service-name bus --git-repo-url $GIT_REPO_URL --pipelines-folder servicescenario" succeeds
```
Comment (amitkrout) [inline test/e2e/features/basic.feature:76]: Pass ```https://github.com/kam-bot/bus.git``` instead of $GIT_REPO_URL

**Review by amitkrout — CHANGES_REQUESTED** [review]

**Review by amitkrout — CHANGES_REQUESTED** [review]
Looks like the CI cluster takes longer time to sync the application. Can you please increase timeout to 3mins, for example ```Wait for "180" seconds```. If it works then we can increase the CI resources and also we have to implement a proper wait logic.

**Review by amitkrout — CHANGES_REQUESTED** [review]

**Comment by varshab1210** [comment]
I am able to reproduce the failure on 4.8 ocp cluster locally

**Comment by amitkrout** [comment]
blocked due to - https://github.com/redhat-developer/kam/pull/266
/block

**Pull request merged**